src/core/ChaosPlan.cpp

```diff
@@ -74,20 +74,35 @@ void ChaosPlan::schedule() {
             dependees.emplace_back(dee->promise_->getFuture());
         }
         auto actionPtr = action.get();
-        folly::collect(dependees)
-                    .via(threadsPool_.get())
-                    .thenValue([actionPtr](auto&&) {
-                        actionPtr->run();
-                    })
-                    .thenError([this, actionPtr](auto ew) {
+
+        folly::collectAll(std::move(dependees))
+            .via(threadsPool_.get())
+            .thenValue([this, actionPtr](const auto&& tries) mutable {
```

Comment:
If a previous action is failed, we could just return directly, we don't need to run current action as well.
We could check if status_ is SUCCEEDED.

Reply:
The same is true when there is.
If action `a` depends on `(b, c)`,   both `b` and `c` need to be executed completely,
Then according to `b` and` c` are all successful, `a` starts to execute

If one of `b` and` c` fails,  `a` is not executed, and it returns directly to failure

Reply:
What I mean is if one of b and c failed, `a` itself should not run anymore. `!hasException` does not mean action succeed, you need to check the action status as well.

Reply:
In Action::run(),  If the action execution fails, exception will be set, Value will be set only when successful.

![image](https://user-images.githubusercontent.com/59197347/109931346-b6ca0980-7d03-11eb-8e67-9df8a5f88ac1.png)

Reply:
I know, as for `a`'s `doRun` should not be called if `b` or `c` failed. Because we only need to know where we fail, continue the plan has no meaning.